plugins/module_utils/network/sonic/config/vlans/vlans.py

```diff
@@ -200,33 +204,31 @@ def _state_overridden(self, want, have, diff):
         commands = []
         requests = []
 
-        r_diff = get_diff(have, want, TEST_KEYS)
-        if not diff and not r_diff:
-            return commands, requests
+        reverse_diff = get_diff(have, want, TEST_KEYS)
 
-        del_vlans = []
-        del_descr_vlans = []
-        for config in r_diff:
-            vlan_obj = search_obj_in_list(config['vlan_id'], want, 'vlan_id')
-            if vlan_obj:
-                if vlan_obj.get('description', None) is None:
-                    del_descr_vlans.append(config)
-            else:
-                del_vlans.append(config)
-
-        if del_vlans:
-            del_requests = self.get_delete_vlans_requests(del_vlans, True)
-            requests.extend(del_requests)
-            commands.extend(update_states(del_vlans, "deleted"))
+        if not diff and not reverse_diff:
+            return commands, requests
 
-        if del_descr_vlans:
-            del_requests = self.get_delete_vlans_requests(del_descr_vlans, False)
+        del_commands = []
+        # Find VLANS to delete or modify using reverse_diff and two lists to separate complete deletion and modification
+        for config in reverse_diff:
+            want_vlan_obj = search_obj_in_list(config['vlan_id'], want, 'vlan_id')
+            if want_vlan_obj:
+                if want_vlan_obj.get('description', None) is None and config.get('description', None) is not None:
+                    del_commands.append({"vlan_id": config.get("vlan_id"), "description": config.get("description")})
+                if want_vlan_obj.get('autostate', None) is True and config.get('autostate', None) is False:
```

Comment:
autostate deletion should also occur here If the reverse_diff vlan has no autostate option, but autostate is configured (to False).

```suggestion
                if ((want_vlan_obj.get('autostate', None) is True or
                     want_vlan_obj.get('autostate', None) is None) and
                    config.get('autostate', None) is False):
```